api/src/test/java/org/openmrs/module/fhir2/api/search/AllergyIntoleranceSearchQueryImplTest.java

```diff
@@ -490,4 +494,86 @@ public void searchForAllergies_shouldHandleComplexQuery() {
 		    equalTo(PATIENT_UUID));
 	}
 	
+	@Test
+	public void shouldReturnCollectionOfAllergiesSortedBySeveritySevere() {
+		
+		SortSpec sort = new SortSpec();
+		sort.setParamName(AllergyIntolerance.SP_SEVERITY);
+		sort.setOrder(SortOrderEnum.ASC);
+		
+		IBundleProvider results = search(
+		 new SearchParameterMap().addParameter(FhirConstants.SEVERITY_SEARCH_HANDLER, FhirConstants.GLOBAL_PROPERTY_SEVERE, new TokenAndListParam().addAnd(new TokenParam(SEVERITY_SEVERE))).setSortSpec(sort));
+		
+		sort.setOrder(SortOrderEnum.DESC);
```

Comment:
This should be *after* the for loop.

Reply:
@ibacher i have fixed it ,,,kindly review